### ISSUE TITLE
Integrals of `CylinderSurface` blocked by upstream issue

### ISSUE THREAD

**Issue by mikeingold** [mikeingold/MeshIntegrals.jl#17]
Integration of a `CylinderSurface` geometry is currently complicated.
- Meshes.jl implements a parametric function whose trace defines only the surface of the tubular component, but it's intended to also represent the surface of the disks at the top and bottom.
- Meshes.jl currently defines `measure(::CylinderSurface)` to calculate surface area (including the top and bottom disks) with the assumption that the cylinder is a right-cylinder with parallel top and bottom planes. https://github.com/JuliaGeometry/Meshes.jl/issues/780
- We define a method `integral(f, cs::CylinderSurface, ::GaussKronrod)` that seems to work well. Unit testing currently compares its results against `measure(cs)` but is required to use a right-cylinder as a test case for the reason stated in the previous bullet point. Results for non-right-cylinders are probably accurate but unverified.
- Methods for integrating with `GaussLegendre` and `GaussKronrod` are currently on hold pending work in Meshes.jl to support parameterizing the top and bottom disks.

**Comment by mikeingold** [comment]
It looked previously like both `Meshes.measure(::Cylinder)` and `Meshes.measure(::CylinderSurface)` methods were built on an assumption of a right-cylinder (top and bottom planes are parallel). Since then, I was able to validate that `measure(::Cylinder)` is accurate for all cylinders whose top and bottom disks do not intersect. I suspect this is also true for surface area, i.e. `measure(::CylinderSurface)`, but have not yet verified this.

**Comment by mikeingold** [comment]
The only remaining issue with this seems to be that the available parameterization function `(::CylinderSurface)(u,v)` only traces the cylinder walls and not the disks at each end. In the short-term, we could probably modify these custom `integral` methods to utilize the standard formula for the walls, but also manually implement parametric functions for the disks.

**Comment by mikeingold** [comment]
Marking this as complete since I've added a specific method that manually integrates all three components of a `CylinderSurface`. However, this may change in a future version of Meshes.jl, where a cylinder surface may become a composite surface object possessing a `CylinderSurface` walls and two `Disk`s.
